ledger/ledger-utxo/src/test/java/net/corda/v5/ledger/utxo/transaction/UtxoFilteredTransactionJavaApiTest.java

```diff
@@ -0,0 +1,66 @@
+package net.corda.v5.ledger.utxo.transaction;
+
+import net.corda.v5.crypto.SecureHash;
+import net.corda.v5.ledger.utxo.Command;
+import net.corda.v5.ledger.utxo.StateAndRef;
+import net.corda.v5.ledger.utxo.StateRef;
+import net.corda.v5.ledger.utxo.TimeWindow;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.util.List;
+
+public class UtxoFilteredTransactionJavaApiTest {
+
+    protected final SecureHash hash = SecureHash.parse("SHA256:0000000000000000000000000000000000000000000000000000000000000000");
+
+    @Test
+    void canReadFilteredTransaction() {
+        // Set up Filtered TX mock
+        UtxoFilteredTransaction filteredTx = Mockito.mock(UtxoFilteredTransaction.class);
+
+        // Notary, time window commands
+        Mockito.when(filteredTx.getNotary()).thenReturn(null);
+        Mockito.when(filteredTx.getTimeWindow()).thenReturn(Mockito.mock(TimeWindow.class));
+        Mockito.when(filteredTx.getCommands()).thenReturn(Mockito.mock(UtxoFilteredData.UtxoFilteredDataRemoved.class));
+
+        // Outputs are size only
+        UtxoFilteredData.UtxoFilteredDataSizeOnly<StateAndRef<?>> outputs
+                = Mockito.mock(UtxoFilteredData.UtxoFilteredDataSizeOnly.class);
+        Mockito.when(outputs.getSize()).thenReturn(5);
+        Mockito.when(filteredTx.getOutputStateAndRefs()).thenReturn(outputs);
+
+        // Inputs is a filtered Audit proof
+        FilteredEntry<StateRef> inputData = Mockito.mock(FilteredEntry.class);
+        Mockito.when(inputData.getIndex()).thenReturn(1);
+        Mockito.when(inputData.getValue()).thenReturn(new StateRef(hash, 0));
+        UtxoFilteredData.UtxoFilteredDataAudit<StateRef> inputs =
+                Mockito.mock(UtxoFilteredData.UtxoFilteredDataAudit.class);
+        Mockito.when(inputs.getSize()).thenReturn(2);
```

Comment:
So is the implication that a filtered audit proof will always imply a size proof? I.e. it looks like this test case is choosing to selectively reveal one input state, but is also declaring a size of 2. I assume there are no use cases where we would want to reveal a subset of the data, but hide the total number?

Reply:
I don't think that can be done - an audit proof of the underling tx always has the size, and needs to give you the position of the revealed fields, so you'd always get a lower bound on the size.